snuba/querylog/__init__.py

```diff
@@ -24,6 +24,7 @@ def _record_timer_metrics(
             "referrer": referrer,
             "parent_api": request.settings.get_parent_api(),
             "final": final,
+            "dataset": query_metadata.dataset,
```

Comment:
You need to add this to the `mark_tags` as well, otherwise you will have the dataset tags only in the top level metric `snuba.api.query` and not in the marks `snuba.api.query.execute`.